### PR TITLE
fix(asset-service): make asset descriptions match user locale

### PR DESCRIPTION
**Pull request by sebfk**
## Description

Currently in lib, we are using the asset descriptions from Coingecko API and getting only the en description from it.
Similarly, our overriden descriptions are only in English.

I did so descriptions are dynamic according to the user language.

Fix to the issue: #836 

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Testing

None.

### CODE REVIEW

**Comment by gomesalexandre** [comment]
@sebfk Please add testing steps for this PR and follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) (it won't fail CI in lib, but should be followed for cleanness and consistency)

File: packages/asset-service/src/service/descriptions.json [hunk 1]
```diff
@@ -1,22 +1,184 @@
...
+    "es": "",
+    "id": "",
+    "ko": "",
+    "pt": "",
+    "zh": "",
+  }
   "eip155:1/erc20:0xdc49108ce5c57bc3408c3a5e95f3d864ec386ed3": "FOXy is a rebasing ERC-20 Token, which you can exchange 1:1 with FOX Tokens, that offer both yield and profit redistribution from the ShapeShift DAO. Get more out of your FOX without losing your governance power.",
   "cosmos:cosmoshub-4/slip44:118": "Cosmos (ATOM) is a cryptocurrency that powers an ecosystem of blockchains designed to scale and interoperate with each other. The Cosmos vision aligns with the ShapeShift vision of a multi chain future, and enables a network of blockchains to be able to communicate with each other in a decentralized manner. Cosmos is a proof-of-stake chain and allows holders to stake their ATOM to earn rewards using ShapeShift.",
   "cosmos:osmosis-1/ibc:27394FB092D2ECCD56123C74F36E4C1F926001CEADA9CA97EA622B25F41E5EB2": "Cosmos (ATOM) on Osmosis is a proof of stake cryptocurrency that powers an ecosystem of blockchains designed to interoperate and communicate with each other in a decentralized manner on the Osmosis blockchain, an alternative decentralized [IBC](https://tendermint.com/ibc/) blockchain."
```
Comment (theoboldfrazier) [inline packages/asset-service/src/service/descriptions.json:184]: These will need to be the same structure as the others.

Also, this structure change will break the existing tests in AssetService.test.ts. The mocked structure for descriptions.json will need to be updated to the new structure.

File: packages/asset-service/src/service/AssetService.ts [hunk 2]
```diff
@@ -29,14 +39,14 @@ export class AssetService {
     if (description) return { description, isTrusted: true }
```
Comment (theoboldfrazier) [inline packages/asset-service/src/service/AssetService.ts:39]: need to add a new parameter to the description method called `locale` of type `LocaleType`.

Also with the new structure of the json, need to update the descriptions type to `Record<string, Record<LocaleType, string>>` and need to change line 36 to

`const description = descriptions[assetId]?.[locale] || descriptions[assetId]?.en || ''`

**Review by gomesalexandre — CHANGES_REQUESTED** [review]
- This is failing lint and type-check, please run `yarn local-ci` locally.
- The PRd diff has invalid syntax on the JSON and is missing commas between members/is having invalid trailing commas after members
- The `locale` variable is undefined

Additionally, I would expect a rationale as to why the `descriptions.json` shape was preferred as a solution, vs. the other proposed (and preferred) solution in the bounty, of using a similar splitting and i18n logic as in web.

A PR with code that has invalid syntax and wouldn't even compile/run is not an acceptable PR for reviewers to spend time on.
Please make sure to take the time on making a tested PR, else we'll have to close this one out.

**Comment by 0xApotheosis** [comment]
Moving to draft until reviewer comments are addressed and CI passes.

File: packages/asset-service/src/service/descriptions.json (hunk 1 above)
Comment (sebfk) [reply packages/asset-service/src/service/descriptions.json]: Done. Should be done now.

File: packages/asset-service/src/service/AssetService.ts (hunk 2 above)
Comment (sebfk) [reply packages/asset-service/src/service/AssetService.ts:39]: Done.

File: packages/asset-service/src/service/AssetService.ts [hunk 3]
```diff
@@ -9,6 +9,16 @@ import localAssetData from './generatedAssetData.json'
 type DescriptionData = Readonly<{ description: string; isTrusted?: boolean }>
 
 export type AssetsById = Record<AssetId, Asset>
+  
+type LocaleType =
```
[the remark this answers is not in the document]
Comment (sebfk) [reply packages/asset-service/src/service/AssetService.ts]: Ran "yarn local-ci" 0 errors.


Done regarding the changes.

File: packages/asset-service/src/service/AssetService.ts [hunk 4]
```diff
@@ -29,14 +39,14 @@ export class AssetService {
...
-      type CoinData = { description: { en: string } }
+      type CoinData = { description: { [key in LocaleType]: string } }
 
       const url = adapters.makeCoingeckoAssetUrl(assetId)
       if (!url) throw new Error()
 
       const { data } = await axios.get<CoinData>(url)
 
-      return { description: data?.description?.en ?? '' }
+      return { description: data?.description?.[locale] || data?.description?.en || '' }
```
[the remark this answers is not in the document]
Comment (sebfk) [reply packages/asset-service/src/service/AssetService.ts:49]: Why do you want it to be ?? over | | do you want it to be:

return { description: data?.description?.[locale] ?? data?.description?.en ?? '' }

File: packages/asset-service/src/service/descriptions.json [hunk 5]
```diff
@@ -1,22 +1,184 @@
...
+    "en": "This token represents a liquidity pool on Curve, a leading decentralized exchange. Holders earn fees from users trading in the pool, and can also deposit the LP token on [ShapeShift](/defi/earn) to earn yield. This liquidity pool contains [CRV](/assets/eip155:1/erc20:0xd533a949740bb3306d119cc777fa900ba034cd52) and [ETH](/assets/eip155:1/slip44:60). To acquire Curve CRV-ETH, deposit [CRV](/assets/eip155:1/erc20:0xd533a949740bb3306d119cc777fa900ba034cd52) and/or [ETH](/assets/eip155:1/slip44:60) on [Curve](https://curve.fi/crveth/deposit). Please be aware that as liquidity pools are composed of differently-priced assets, they are subject to impermanent loss.",
+    "fr": "",
+    "ru": "",
+    "es": "",
+    "id": "",
+    "ko": "",
+    "pt": "",
+    "zh": "",
+  }
   "eip155:1/erc20:0xdc49108ce5c57bc3408c3a5e95f3d864ec386ed3": "FOXy is a rebasing ERC-20 Token, which you can exchange 1:1 with FOX Tokens, that offer both yield and profit redistribution from the ShapeShift DAO. Get more out of your FOX without losing your governance power.",
```
[the remark this answers is not in the document]
Comment (sebfk) [reply packages/asset-service/src/service/descriptions.json]: Updated.

File: packages/asset-service/src/service/descriptions.json [hunk 6]
```diff
@@ -1,22 +1,184 @@
...
+  "eip155:1/erc20:0xed4064f376cb8d68f770fb1ff088a3d0f3ff5c4d": {
+    "en": "This token represents a liquidity pool on Curve, a leading decentralized exchange. Holders earn fees from users trading in the pool, and can also deposit the LP token on [ShapeShift](/defi/earn) to earn yield. This liquidity pool contains [CRV](/assets/eip155:1/erc20:0xd533a949740bb3306d119cc777fa900ba034cd52) and [ETH](/assets/eip155:1/slip44:60). To acquire Curve CRV-ETH, deposit [CRV](/assets/eip155:1/erc20:0xd533a949740bb3306d119cc777fa900ba034cd52) and/or [ETH](/assets/eip155:1/slip44:60) on [Curve](https://curve.fi/crveth/deposit). Please be aware that as liquidity pools are composed of differently-priced assets, they are subject to impermanent loss.",
+    "fr": "",
+    "ru": "",
+    "es": "",
+    "id": "",
+    "ko": "",
+    "pt": "",
+    "zh": "",
```
[the remark this answers is not in the document]
Comment (sebfk) [reply packages/asset-service/src/service/descriptions.json]: Updated.

File: packages/asset-service/src/service/descriptions.json [hunk 7]
```diff
@@ -1,22 +1,184 @@
...
+  }
```
[the remark this answers is not in the document]
Comment (sebfk) [reply packages/asset-service/src/service/descriptions.json]: Updated.

File: packages/asset-service/src/service/descriptions.json [hunk 8]
```diff
@@ -1,22 +1,184 @@
...
+  "eip155:1/erc20:0xca3d75ac011bf5ad07a98d02f18225f9bd9a6bdf": {
+    "en": "This token represents a liquidity pool on Curve, a leading decentralized exchange. Holders earn fees from users trading in the pool, and can also deposit the LP token on [ShapeShift](/defi/earn) to earn yield. This liquidity pool contains [CRV](/assets/eip155:1/erc20:0xd533a949740bb3306d119cc777fa900ba034cd52) and [ETH](/assets/eip155:1/slip44:60). To acquire Curve CRV-ETH, deposit [CRV](/assets/eip155:1/erc20:0xd533a949740bb3306d119cc777fa900ba034cd52) and/or [ETH](/assets/eip155:1/slip44:60) on [Curve](https://curve.fi/crveth/deposit). Please be aware that as liquidity pools are composed of differently-priced assets, they are subject to impermanent loss.",
+    "fr": "",
+    "ru": "",
+    "es": "",
+    "id": "",
+    "ko": "",
+    "pt": "",
+    "zh": "",
+  }
```
[the remark this answers is not in the document]
Comment (sebfk) [reply packages/asset-service/src/service/descriptions.json]: Updated.

File: packages/asset-service/src/service/descriptions.json [hunk 9]
```diff
@@ -1,22 +1,184 @@
...
+  "eip155:1/erc20:0x9d0464996170c6b9e75eed71c68b99ddedf279e8": {
+    "en": "This token represents a liquidity pool on Curve, a leading decentralized exchange. Holders earn fees from users trading in the pool, and can also deposit the LP token on [ShapeShift](/defi/earn) to earn yield. This liquidity pool contains [CRV](/assets/eip155:1/erc20:0xd533a949740bb3306d119cc777fa900ba034cd52) and [ETH](/assets/eip155:1/slip44:60). To acquire Curve CRV-ETH, deposit [CRV](/assets/eip155:1/erc20:0xd533a949740bb3306d119cc777fa900ba034cd52) and/or [ETH](/assets/eip155:1/slip44:60) on [Curve](https://curve.fi/crveth/deposit). Please be aware that as liquidity pools are composed of differently-priced assets, they are subject to impermanent loss.",
+    "fr": "",
+    "ru": "",
+    "es": "",
+    "id": "",
+    "ko": "",
+    "pt": "",
+    "zh": "",
+  }
```
[the remark this answers is not in the document]
Comment (sebfk) [reply packages/asset-service/src/service/descriptions.json]: Updated.

File: packages/asset-service/src/service/descriptions.json [hunk 10]
```diff
@@ -1,22 +1,184 @@
...
+  "eip155:1/erc20:0x194ebd173f6cdace046c53eacce9b953f28411d1": {
+    "en": "This token represents a liquidity pool on Curve, a leading decentralized exchange. Holders earn fees from users trading in the pool, and can also deposit the LP token on [ShapeShift](/defi/earn) to earn yield. This liquidity pool contains [CRV](/assets/eip155:1/erc20:0xd533a949740bb3306d119cc777fa900ba034cd52) and [ETH](/assets/eip155:1/slip44:60). To acquire Curve CRV-ETH, deposit [CRV](/assets/eip155:1/erc20:0xd533a949740bb3306d119cc777fa900ba034cd52) and/or [ETH](/assets/eip155:1/slip44:60) on [Curve](https://curve.fi/crveth/deposit). Please be aware that as liquidity pools are composed of differently-priced assets, they are subject to impermanent loss.",
+    "fr": "",
+    "ru": "",
+    "es": "",
+    "id": "",
+    "ko": "",
+    "pt": "",
+    "zh": "",
+  }
```
[the remark this answers is not in the document]
Comment (sebfk) [reply packages/asset-service/src/service/descriptions.json]: Updated.

File: packages/asset-service/src/service/descriptions.json [hunk 11]
```diff
@@ -1,22 +1,184 @@
...
+  "eip155:1/erc20:0xb9446c4ef5ebe66268da6700d26f96273de3d571": {
+    "en": "This token represents a liquidity pool on Curve, a leading decentralized exchange. Holders earn fees from users trading in the pool, and can also deposit the LP token on [ShapeShift](/defi/earn) to earn yield. This liquidity pool contains [CRV](/assets/eip155:1/erc20:0xd533a949740bb3306d119cc777fa900ba034cd52) and [ETH](/assets/eip155:1/slip44:60). To acquire Curve CRV-ETH, deposit [CRV](/assets/eip155:1/erc20:0xd533a949740bb3306d119cc777fa900ba034cd52) and/or [ETH](/assets/eip155:1/slip44:60) on [Curve](https://curve.fi/crveth/deposit). Please be aware that as liquidity pools are composed of differently-priced assets, they are subject to impermanent loss.",
+    "fr": "",
+    "ru": "",
+    "es": "",
+    "id": "",
+    "ko": "",
+    "pt": "",
+    "zh": "",
+  }
```
[the remark this answers is not in the document]
Comment (sebfk) [reply packages/asset-service/src/service/descriptions.json]: Updated.

File: packages/asset-service/src/service/descriptions.json [hunk 12]
```diff
@@ -1,22 +1,184 @@
...
+  "eip155:1/erc20:0x3a283d9c08e8b55966afb64c515f5143cf907611": {
+    "en": "This token represents a liquidity pool on Curve, a leading decentralized exchange. Holders earn fees from users trading in the pool, and can also deposit the LP token on [ShapeShift](/defi/earn) to earn yield. This liquidity pool contains [CRV](/assets/eip155:1/erc20:0xd533a949740bb3306d119cc777fa900ba034cd52) and [ETH](/assets/eip155:1/slip44:60). To acquire Curve CRV-ETH, deposit [CRV](/assets/eip155:1/erc20:0xd533a949740bb3306d119cc777fa900ba034cd52) and/or [ETH](/assets/eip155:1/slip44:60) on [Curve](https://curve.fi/crveth/deposit). Please be aware that as liquidity pools are composed of differently-priced assets, they are subject to impermanent loss.",
+    "fr": "",
+    "ru": "",
+    "es": "",
+    "id": "",
+    "ko": "",
+    "pt": "",
+    "zh": "",
+  }
```
[the remark this answers is not in the document]
Comment (sebfk) [reply packages/asset-service/src/service/descriptions.json]: Updated.

**Comment by sebfk** [comment]
Done fixes. Only thing is regarding the Nullish coalescing operator.

File: packages/asset-service/src/service/descriptions.json (hunk 1 above)
Comment (gomesalexandre) [reply packages/asset-service/src/service/descriptions.json]: Is it? There is no mocks update in this diff

File: packages/asset-service/src/service/AssetService.ts (hunk 2 above)
Comment (gomesalexandre) [reply packages/asset-service/src/service/AssetService.ts:39]: Is it? There is no tests update in this diff

File: packages/asset-service/src/service/AssetService.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (gomesalexandre) [reply packages/asset-service/src/service/AssetService.ts]: That doesn't look like 0 errors to me https://github.com/shapeshift/lib/runs/7104723210?check_suite_focus=true
Lint and test scripts are breaking

**Pull request closed without merging**